compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -12619,6 +12623,103 @@ private STNode parseListMatchPatternMemberRhs(SyntaxKind nextTokenKind) {
         }
     }
 
+    /**
+     * Parse mapping match pattern.
+     * <p>
+     *     mapping-match-pattern := { field-match-patterns }
+     *     field-match-patterns :=
+     *    field-match-pattern (, field-match-pattern)* [, rest-match-pattern]
+     *    | [ rest-match-pattern ]
+     *      field-match-pattern := field-name : match-pattern
+     *      rest-match-pattern := ... var variable-name
+     * </p>
+     *
+     * @return Parsed Node.
+     */
+    private STNode parseMappingMatchPattern() {
+        startContext(ParserRuleContext.MAPPING_MATCH_PATTERN);
+        STNode openBraceToken = parseOpenBrace();
+        List<STNode> mappingMatchPatternList = new ArrayList<>();
+        STNode restMatchPattern = null;
+
+        while (!isEndOfMappingMatchPattern()) {
+            STToken nextToken = peek();
+            if (nextToken.kind == SyntaxKind.ELLIPSIS_TOKEN) {
+                restMatchPattern = parseRestMatchPattern();
```

Comment:
I think we should first parse the match-pattern, and then check whether its a rest-pattern or not.

Reason is, if theres a junk token before ellipsis, above condition fail and it will try to parse it as a field-match-pattern.

Reply:
Also, once the rest pattern is found, we should continue to parse match-patterns (do not break the loop), but add them as invalid nodes.

Reply:
Thanks for mentioning it. I will change.

Reply:
changed